source/val/validate_image.cpp

```diff
@@ -1122,6 +1132,14 @@ spv_result_t ValidateImageLod(ValidationState_t& _, const Instruction* inst) {
 
   if (spv_result_t result = ValidateImageCommon(_, inst, info)) return result;
 
+  if (info.multisampled) {
```

Comment:
Performing this check early simplifies validation. We no longer have to cross-check Lod vs. Sample, or Grad vs. Sample.

This lets us delete a few tests because we've already rejected sampling operations for multisampled images.

Reply:
I implemented the multisampled check in each class of instruction: sampling, dref sampling, and gather.
That way I can give a more specific error message.